onadata/apps/api/tests/viewsets/test_team_viewset.py

```diff
@@ -174,9 +180,42 @@ def test_team_share(self):
         view = TeamViewSet.as_view({"post": "share"})
 
         ROLES = [ReadOnlyRole, EditorRole]
+        ROLES_SET = [ReadOnlyRoleNoDownload, DataEntryOnlyRole]
 
-        for role_class in ROLES:
+        for role_class, role_class_set_for_user in zip(ROLES, ROLES_SET):
             self.assertFalse(role_class.user_has_role(user_chuck, self.project))
+            self.assertFalse(
+                role_class_set_for_user.user_has_role(user_chuck, self.project)
+            )
+            data = {"role": role_class.name, "project": self.project.pk}
+            request = self.factory.post(
+                "/",
+                data=json.dumps(data),
+                content_type="application/json",
+                **self.extra,
+            )
+            response = view(request, pk=self.team.pk)
+
+            self.assertEqual(response.status_code, 204)
+
+            self.assertTrue(role_class.user_has_role(user_chuck, self.project))
+            self.assertTrue(
+                role_class_set_for_user.user_has_role(user_chuck, self.xform)
+            )
+        metadata = self.xform.metadata_set.get(data_type="xform_meta_perms")
+        serializer = MetaDataSerializer(
+            metadata,
+            data={
+                "data_value": "editor|dataentry|readonly",
+                "data_type": "xform_meta_perms",
+                "xform": self.xform.id,
+            },
+        )
+
+        if serializer.is_valid():
```

Comment:
We can omit serializer.is_valid() since for tests it doesn't matter if an exception is thrown